app/code/Magento/ConfigurableProduct/Model/ResourceModel/Attribute/OptionSelectBuilder.php

```diff
@@ -29,14 +30,25 @@ class OptionSelectBuilder implements OptionSelectBuilderInterface
      */
     private $attributeOptionProvider;
 
+    /**
+     * Eav Attribute Resource Model.
+     *
+     * @var EavAttributeResource
+     */
+    private $eavAttributeResource;
+
     /**
      * @param Attribute $attributeResource
      * @param OptionProvider $attributeOptionProvider
      */
-    public function __construct(Attribute $attributeResource, OptionProvider $attributeOptionProvider)
-    {
+    public function __construct(
+        Attribute $attributeResource,
+        OptionProvider $optionProvider,
```

Comment:
Please revert the parameter name. It is breaking change. You may ignore Codacy alerts.